employee_scheduling.py

```diff
@@ -0,0 +1,204 @@
+from dimod import (
+    Binary,
+    BinaryQuadraticModel,
+    ConstrainedQuadraticModel,
+    quicksum,
+)
+from dwave.system import LeapHybridCQMSampler
+
+
+def build_cqm(
+    availability,
+    shifts,
+    min_shifts,
+    max_shifts,
+    shift_min,
+    shift_max,
+    manager,
+    isolated,
+    k,
+):
+    cqm = ConstrainedQuadraticModel()
+    employees = [key for key in availability.keys()]
```

Comment:
In fact, if `availability` is a `dict` (as it seems to be):
```suggestion
    employees = list(availability)
```